src/snowflake/connector/connection.py

```diff
@@ -913,100 +925,108 @@ def __open_connection(self):
 
         # Setup authenticator
         auth = Auth(self.rest)
-        if self.auth_class is not None:
-            if type(
-                self.auth_class
-            ) not in FIRST_PARTY_AUTHENTICATORS and not issubclass(
-                type(self.auth_class), AuthByKeyPair
-            ):
-                raise TypeError("auth_class must be a child class of AuthByKeyPair")
-                # TODO: add telemetry for custom auth
-            self.auth_class = self.auth_class
-        elif self._authenticator == DEFAULT_AUTHENTICATOR:
-            self.auth_class = AuthByDefault(
-                password=self._password,
-                timeout=self._login_timeout,
-                backoff_generator=self._backoff_generator,
-            )
-        elif self._authenticator == EXTERNAL_BROWSER_AUTHENTICATOR:
-            self._session_parameters[PARAMETER_CLIENT_STORE_TEMPORARY_CREDENTIAL] = (
-                self._client_store_temporary_credential if IS_LINUX else True
-            )
-            auth.read_temporary_credentials(
-                self.host,
-                self.user,
-                self._session_parameters,
+
+        if self._session_token and self._master_token:
+            auth._rest.update_tokens(
```

Comment:
I think we should test whether the "authentication" was successful somehow

Reply:
Okay, if I try to make a heartbeat call there and then it fails, user will get this error in their terminal for example
```
Traceback (most recent call last):
  File "/Users/klin/git/sp-test/src/sp_test.py", line 3, in <module>
    conn = snowflake.connector.connect(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/__init__.py", line 54, in Connect
    return SnowflakeConnection(**kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/connection.py", line 425, in __init__
    self.connect(**kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/connection.py", line 715, in connect
    self.__open_connection()
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/connection.py", line 935, in __open_connection
    ret = auth._rest._heartbeat()
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 609, in _heartbeat
    ret = self._post_request(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 730, in _post_request
    ret = self.fetch(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 839, in fetch
    ret = self._request_exec_wrapper(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 975, in _request_exec_wrapper
    raise e
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 880, in _request_exec_wrapper
    return_object = self._request_exec(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 1189, in _request_exec
    raise err
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 1131, in _request_exec
    raise_failed_request_error(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/network.py", line 238, in raise_failed_request_error
    Error.errorhandler_wrapper(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/errors.py", line 290, in errorhandler_wrapper
    handed_over = Error.hand_to_other_handler(
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/errors.py", line 348, in hand_to_other_handler
    connection.errorhandler(connection, cursor, error_class, error_value)
  File "/opt/homebrew/lib/python3.9/site-packages/snowflake/connector/errors.py", line 221, in default_errorhandler
    raise error_class(
snowflake.connector.errors.InterfaceError: 250003 (08001): 401 Unauthorized: post https://ig86763.preprod12.us-west-2.aws.snowflakecomputing.com:443/session/heartbeat?requestId=4a63f07e-4928-4f98-b02d-58a4e7151f73&request_guid=148cac21-a4da-4341-bb24-862f8a6211c7
```

Does that seem like a reasonable experience to give a dev for when the tokens are invalid? Should that error get caught and replaced with clearer wording?